src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2436,118 +2476,312 @@ public void UpdateIsNullableEnabledForConstructorsAndFields(bool useStatic, bool
+            {
+                var instanceInitializers = InstanceInitializersForPositionalMembers is null
+                    ? declaredMembers.InstanceInitializers
+                    : mergeInitializers();
 
-                foreach (FieldOrPropertyInitializer.Builder initializer in group)
-                {
-                    builder.Add(new FieldOrPropertyInitializer(initializer.FieldOpt, initializer.Syntax, syntaxLength));
+                return new MembersAndInitializers(
+                    NonTypeNonIndexerMembers.ToImmutableAndFree(),
+                    declaredMembers.StaticInitializers,
+                    instanceInitializers,
+                    declaredMembers.IndexerDeclarations,
+                    isNullableEnabledForInstanceConstructorsAndFields: IsNullableEnabledForInstanceConstructorsAndFields,
+                    isNullableEnabledForStaticConstructorsAndFields: IsNullableEnabledForStaticConstructorsAndFields);
 
-                    // A constant field of type decimal needs a field initializer, so
-                    // check if it is a metadata constant, not just a constant to exclude
-                    // decimals. Other constants do not need field initializers.
-                    if (initializer.FieldOpt == null || !initializer.FieldOpt.IsMetadataConstant)
+                ImmutableArray<ImmutableArray<FieldOrPropertyInitializer>> mergeInitializers()
+                {
+                    var groupCount = declaredMembers.InstanceInitializers.Length;
+                    var groupsBuilder = ArrayBuilder<ImmutableArray<FieldOrPropertyInitializer>>.GetInstance(groupCount);
+                    for (int i = 0; i < groupCount; i++)
                     {
-                        // ignore leading and trailing trivia of the node:
-                        syntaxLength += initializer.Syntax.Span.Length;
+                        if (i == declaredMembers.InstanceInitializersIndexForRecordDeclarationWithParameters)
+                        {
+                            var insertedInitializers = InstanceInitializersForPositionalMembers;
+                            var declaredInitializers = declaredMembers.InstanceInitializers[i];
+                            if (declaredInitializers.Length > 0)
+                            {
+                                // initializers should be added in syntax order:
+                                Debug.Assert(insertedInitializers[insertedInitializers.Count - 1].Syntax.SyntaxTree == declaredInitializers[0].Syntax.SyntaxTree);
+                                Debug.Assert(insertedInitializers[insertedInitializers.Count - 1].Syntax.Span.Start < declaredInitializers[0].Syntax.Span.Start);
+                            }
+
+                            var builder = ArrayBuilder<FieldOrPropertyInitializer>.GetInstance(insertedInitializers.Count + declaredInitializers.Length);
```

Comment:
>var builder = ArrayBuilder<FieldOrPropertyInitializer>.GetInstance(insertedInitializers.Count + declaredInitializers.Length); [](start = 28, length = 125)

If we are going to free ```InstanceInitializersForPositionalMembers``` anyway, why not simply add to the end of it instead of creating a new builder and performing an extra copy of content?

Reply:
Makes sense. Thanks

---
In reply to: [562958378](https://github.com/dotnet/roslyn/pull/50650#discussion_r562958378) [](ancestors = 562958378)